bin/migrate-mongo-to-kafka.js

```diff
@@ -0,0 +1,157 @@
+const { normalizeChannel } = require('bat-utils/lib/extras-utils')
+const _ = require('underscore')
+const { Runtime } = require('bat-utils')
+const transaction = require('../eyeshade/lib/transaction')
+const referrals = require('../eyeshade/lib/referrals')
+const settlements = require('../eyeshade/lib/settlements')
+const config = require('../config')
+
+const originalRateId = '71341fc9-aeab-4766-acf0-d91d3ffb0bfa'
+config.postgres.url = process.env.MIGRATE_DATABASE_URL
+config.postgres.schemaVersionCheck = false
+process.env.NODE_ENV = 'production'
+config.database.mongo = process.env.MIGRATE_MONGODB_URI
+
+const rconf = _.pick(config, ['kafka', 'currency', 'postgres', 'database'])
+const runtime = new Runtime(rconf)
+
+main().catch(console.error)
+
+async function main () {
+  try {
+    await runtime.kafka.producer()
+    await Promise.all([
+      transferFailedReferrals(),
+      transferFailedSettlements()
+    ])
+  } catch (e) {
+    console.error(e)
+  } finally {
+    await runtime.quit()
+  }
+}
+
+async function transferFailedReferrals () {
+  return connectToKafka('referrals', 'transactionId', referrals, async (queried) => {
+    const ids = queried.map(({
+      publisher,
+      transactionId
+    }) => {
+      const normalizedChannel = normalizeChannel(publisher)
+      return transaction.id.referral(transactionId, normalizedChannel)
+    })
+    const { rows } = await runtime.postgres.query(`
+    select count(*)
+    from transactions
+    where id = any($1::uuid[])`, [ids], true)
+    console.log('count', rows[0].count)
+    if (!(+rows[0].count)) {
+      return []
+    }
+    return queried.map(({
+      downloadId,
+      downloadTimestamp,
+      finalized,
+      referralCode = '',
+      owner,
+      publisher,
+      transactionId,
+      groupId,
+      platform
+    }) => ({
+      downloadId,
+      downloadTimestamp: new Date((downloadTimestamp || finalized).toISOString()),
+      finalizedTimestamp: new Date(finalized.toISOString()),
```

Comment:
should both of these `toISOString()` calls be on the date object?

Reply:
fixed in: eec22ca